sktime/annotation/stray.py

```diff
@@ -0,0 +1,268 @@
+from typing import Dict
+
+import numpy as np
+import numpy.typing as npt
+from sklearn.neighbors import NearestNeighbors
+
+from sktime.transformations.base import BaseTransformer
+
+__author__ = ["KatieBuc"]
+__all__ = ["STRAY"]
+
+
+class STRAY(BaseTransformer):
+    """STRAY: robust anomaly detection in data streams with concept drift.
+
+    This is based on STRAY (Search TRace AnomalY) _[1], which is a modification
+    of HDoutliers _[2]. HDoutliers is a powerful algorithm for the detection of
+    anomalous observations in a dataset, which has (among other advantages) the
+    ability to detect clusters of outliers in multi-dimensional data without
+    requiring a model of the typical behavior of the system. However, it suffers
+    from some limitations that affect its accuracy. STRAY is an extension of
+    HDoutliers that uses extreme value theory for the anomolous threshold
+    calculation, to deal with data streams that exhibit non-stationary behavior.
+
+    Parameters
+    ----------
+    alpha : float, optional (default=0.01)
+        Threshold for determining the cutoff for outliers. Observations are
+        considered outliers if they fall in the (1 - alpha) tail of
+        the distribution of the nearest-neighbor distances between exemplars.
+    k : int, optional (default=10)
+        Number of neighbours considered.
+    knn_algorithm : str {"auto", "ball_tree", "kd_tree", "brute"}, optional
+        (default="brute")
+        Algorithm used to compute the nearest neighbors, from
+        sklearn.neighbors.NearestNeighbors
+    p : float, optional (default=0.5)
+        Proportion of possible candidates for outliers. This defines the starting point
+        for the bottom up searching algorithm.
+    size_threshold : int, optional (default=50)
+        Sample size to calculate an emperical threshold.
+    outlier_tail : str {"min", "max"}, optional (default="max")
+        Direction of the outlier tail.
+
+    Attributes
+    ----------
+    score_ : pd.Series
+        Outlier score of each data point in X.
+    y_ : pd.Series
+        Outlier boolean flag for each data point in X.
+
+    References
+    ----------
+    .. [1] Talagala, Priyanga Dilini, Rob J. Hyndman, and Kate Smith-Miles.
+    "Anomaly detection in high-dimensional data." Journal of Computational
+    and Graphical Statistics 30.2 (2021): 360-374.
+    .. [2] Wilkinson, Leland. "Visualizing big data outliers through
+    distributed aggregation." IEEE transactions on visualization and
+    computer graphics 24.1 (2017): 256-266.
+
+    Examples
+    --------
+    >>> from sktime.annotation.stray import STRAY
+    >>> from sktime.datasets import load_airline
+    >>> from sklearn.preprocessing import MinMaxScaler
+    >>> import numpy as np
+    >>> X = load_airline().head(10)
+    >>> X = np.array(X).reshape(-1, 1)
+    >>> scaler = MinMaxScaler()
+    >>> X = scaler.fit_transform(X)
+    >>> model = STRAY(k=3)
+    >>> y = model.fit_transform(X)
```

Comment:
can we put `MinMaxScaler` and `STRAY` into a `TransformerPipeline` here? Then you dont have type issues and you dont need to reshape? `MinMaxScaler` would have to be used via `TabularToSeriesAdopter`. Its best in docstring example not to do any `numpy `operations in my opinion.

Reply:
creating a pipeline seems a bit  of an overkill for a basic example, how about
```python
X = load_airline().to_frame().to_numpy()
```
That creates a column vector.